tracing-error/src/backtrace.rs

```diff
@@ -68,6 +68,11 @@ pub struct SpanTrace {
 // === impl SpanTrace ===
 
 impl SpanTrace {
+    /// Create a new span trace with the given span as the innermost span.
+    pub fn new(span: Span) -> Self {
```

Comment:
we may also want to add a `From<Span> for SpanTrace` conversion, but we can do that separately.

Reply:
I actually have _a_ reason for not doing so here, though I'm not super certain in the justification:

A `SpanTrace` is suitably different from `Span` that I'm not sure that `From` is clear on the semantics of creating the span trace. While `SpanTrace` physically only holds onto the single `Span`, logically the `SpanTrace` is the whole list of spans, from the innermost span that it holds, to the outermost span. As such, `.into()` feels a bit wrong.

I think it's fine either way, but that's my reasoning for just providing `::new` here.

Reply:
yeah, i was on the fence about it — we can always do that separately, if we decide it's worth having? this seems good to me as is.